src/components/GoalDependencyForm/GoalDependencyForm.tsx

```diff
@@ -41,8 +41,9 @@ interface GoalDependencyAddFormProps {
 }
 
 export const GoalDependencyAddForm: React.FC<GoalDependencyAddFormProps> = ({ goalId, kind, isEmpty, onSubmit }) => {
-    const [selected, setSelected] = useState<GoalByIdReturnType | null>(null);
     const [query, setQuery] = useState('');
+    const [selectedGoal, setSelectedGoal] = useState<GoalByIdReturnType | null>(null);
```

Comment:
Could we drop this useState and replace it with `isGoalSelected`?

```js
const values = getValues();
const isGoalSelected = !!values["relation.id"];
```

Reply:
But you need to add useMemo for isGoalSelected

Reply:
and getValues => [watch](https://react-hook-form.com/docs/useform/watch) wo useMemo

Reply:
```js
const isGoalSelected = !!watch("relation.id")
```